### PR TITLE
MongoDAO - Added NanoLogger for unknown property support

### PR DESCRIPTION
**Pull request by TanayParikh**
**Note:** Documents of unknown property/class still need to be parsed to ensure the parser remains in a valid state. As such null checks for class/prop are needed.

### CODE REVIEW

File: src/foam/dao/MongoDAO.java [hunk 1]
```diff
@@ -112,8 +115,8 @@ public Sink select_(X x, Sink sink, long skip, long limit, Comparator order, Pre
     return sink;
   }
 
-  private FObject createFObject(X x, BsonDocumentReader reader, Class cls) {
-    FObject obj = (FObject) x.create(cls);
+  private FObject createFObject(X x, BsonDocumentReader reader, Class cls, NanoLogger logger) {
+    FObject obj = ( cls == null ) ? null : (FObject) x.create(cls);
```
Comment (adamvy) [inline src/foam/dao/MongoDAO.java:119]: If cls is null that should probably be a hard error.

File: src/foam/dao/MongoDAO.java [hunk 2]
```diff
@@ -185,7 +196,11 @@ private Object getValue(X x, BsonDocumentReader reader, Class cls) {
...
       case DOCUMENT:
-        value = createFObject(x, reader, cls);
+        value = createFObject(x, reader, cls, logger);
+        break;
+
+      case OBJECT_ID:
+        value = reader.readObjectId();
         break;
 
       default:
```
Comment (adamvy) [inline src/foam/dao/MongoDAO.java:206]: Use logger for the default case here as well.

File: src/foam/dao/MongoDAO.java (hunk 1 above)
Comment (TanayParikh) [reply src/foam/dao/MongoDAO.java:119]: This was to handle the case when the mongo field name read in isn't recognized. In that case the value still needs to be read (and ignored) in order to ensure the parser state is maintained. `cls` can be null in the case the unknown field is a nested document. Here, if `cls` is null, it continues parsing through the nested document and just ignores the values (thus `cls` wouldn't be needed).

The `cls` we care about would've been fetched from the DAO's `of` property on line 103.

File: src/foam/dao/MongoDAO.java (hunk 2 above)
Comment (TanayParikh) [reply src/foam/dao/MongoDAO.java:206]: Updated, thanks.

File: src/foam/dao/MongoDAO.java (hunk 1 above)
Comment (adamvy) [reply src/foam/dao/MongoDAO.java:119]: Ah I understand, sounds good to me.

**Pull request merged**